### ISSUE TITLE
How to ModelCheckpoint with custom metrics

### ISSUE THREAD

**Issue by hrdxwandg** [keras-team/keras#11440]
I implement a custom f1 score metric with Callback.
How can I use it to monitor the best model with ModelCheckpoint. So I want save the best model with high f1 score in validat data.
`model_name = "weights.{epoch:03d}-{val_f1:.4f}.hdf5"
checkpoint = ModelCheckpoint(filepath=os.path.join(model_dir, model_name),
                             monitor="val_f1",
                             verbose=1,
                             save_best_only=True,
                             save_weights_only=True,
                             mode='max')`

`model.compile(optimizer=adam, loss='categorical_crossentropy', metrics=[f1])
model.fit(X_train,
              y_train,
              batch_size=batch_size,
              epochs=epochs,
              verbose=1,
              callbacks=[checkpoint, tbCallBack, metrics],
              validation_data=(X_test, y_test))  # starts training`

**Comment by harshini-gadige** [comment]
Hi

This issue isn't related to a bug/enhancement/feature-request or other accepted types of issue.

To ask questions, please see the following resources :

https://keras.io/
https://keras.io/#support
https://gitter.im/Keras-io/Lobby

Thanks!

**Comment by ghayth82** [comment]
I am facing the same issue? is there any workaround, please?
